### PR TITLE
adding date validator and normaliser

### PR DESCRIPTION
**Pull request by kimessenbergfonterra**

### CODE REVIEW

**Comment by kimessenbergfonterra** [comment]
Solved #2

File: transformers.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return date_input.date()
+    elif type(date_input) == str: 
+        return parse(date_input).date()
+    elif type(date_input) == int:
+        return dt.date.fromtimestamp(date_input / 1e3)
+    elif type(date_input) == dt.date:
+        return date_input
+    
+
+
```
Comment (jTaylo99) [inline transformers.py]: Got a lot of empty lines leftover here, try and end the file with only one empty line

File: validation.py [hunk 2]
```diff
@@ -89,3 +91,30 @@ def validate(self, value):
...
+            msg = f'Expected {value!r} to be no more than {self.latest_date!r}'
+            logging.error(msg)
+            raise ValueError(msg)
+        if self.first_of_month and value.day != 1:
+            msg = f'Expected {value!r} to be first day of the month'
+            logging.error(msg)
+            raise ValueError(msg)
+
+
+
```
Comment (jTaylo99) [inline validation.py]: Same thing here about empty lines

File: validation.py [hunk 3]
```diff
@@ -89,3 +91,30 @@ def validate(self, value):
...
+        self.latest_date = latest_date
+        self.first_of_month = first_of_month
+
+    def validate(self, value):
+        if not isinstance(value, (dt.date)):
+            msg = f'Expected {value!r} to be an int or float'
+            logging.error(msg)
+            raise TypeError(msg)
+        if self.earliest_date is not None and value < self.earliest_date:
+            msg = f'Expected {value!r} to be at least {self.earliest_date!r}'
```
Comment (jTaylo99) [inline validation.py]: Couple things. Try use double quotes for all your strings, and I think putting '!r' at the end of the variable will break the message.

**Review by jTaylo99 — CHANGES_REQUESTED** [review]
Some minor formatting issues, but should be fine to merge once those are sorted.

**Comment by jTaylo99** [comment]
Nice, merged!

**Pull request merged**